test/versioned/redis/redis.tap.js

```diff
@@ -24,35 +24,27 @@ test('Redis instrumentation', { timeout: 20000 }, function (t) {
   let client
 
   t.beforeEach(function () {
-    return new Promise((resolve, reject) => {
-      helper.flushRedisDb(DB_INDEX, (error) => {
-        if (error) {
-          reject(error)
+    return new Promise(async (resolve, reject) => {
```

Comment:
For these, I would have changed to have the beforeEach function be an `async` function and then wrapped any additional relevant parts that needed to be promisified instead of passing an async function into a promise constructor. This is sort of an anti pattern as control flow and error handling can get messy and result in unexpected bugs.

That being said, this particular usage I don't think its a big deal to have for a bit as I'm reasonably sure the error would still bubble out to fail the setup appropriately. You would just have problems attempting to handle the error gracefully.

I'll leave up to you if you want to tackle with this or if we want to follow up. We may just want to wait a super long time as not to have the pattern get copied in the future.

Reply:
ok i'll take a look. regardless i will hold on merging until we ship 9.0.0